lib/framework/secrets.dart

```diff
@@ -0,0 +1,39 @@
+// Environment Configuration
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+
+import '../ensemble.dart';
+
+class SecretsStore with Invokable {
+  // ignore since we override getProperty
+  @override
+  Map<String, Function> getters() {
+    throw UnimplementedError();
+  }
+
+  @override
+  getProperty(prop) {
```

Comment:
@vusters I'm not sure how I should handle Future values when reading back from secure storage. Any tips?

Reply:
All getters are synchronous - exposing async is tricky for the user to use. Can we simply read all the secrets into memory and getProperty(key) returns synchronously? If the secrets are modified from the server (local overrides are static) we can simply update the map.

Reply:
So secrets are resolved in the following way?

1. Fetch from local, remote, secure storage
2. Encrypt and store in secure storage (if necessary)
3. Keep copy in memory
4. Secrets read from memory

What value is secure storage adding here?